src/core/utils/hooks/useResizeObserver.tsx

```diff
@@ -6,29 +6,33 @@ import React from 'react';
 
 /**
  * Hook that uses `ResizeObserver` to return an element's size every time it updates.
- * @param elementRef ref of the element to observe resizes on.
+ * @param element the element to observe resizes on. You can pass `.current` member of the element's ref.
  * @returns stateful object containing height and width of the element.
+ *
+ * @example
+ * const elementRef = React.useRef<HTMLDivElement>(null);
+ * const { height, width } = useResizeObserver(elementRef.current);
+ * ...
+ * return <div ref={elementRef}>...</div>;
  */
-export const useResizeObserver = <T extends HTMLElement>(
-  elementRef: React.RefObject<T>,
-) => {
+export const useResizeObserver = <T extends HTMLElement>(element: T | null) => {
   const [size, setSize] = React.useState(() => ({
-    height: elementRef.current?.getBoundingClientRect().height,
-    width: elementRef.current?.getBoundingClientRect().width,
+    height: element?.getBoundingClientRect().height,
+    width: element?.getBoundingClientRect().width,
   }));
 
   const resizeObserver = React.useRef<ResizeObserver | null>(null);
 
   React.useLayoutEffect(() => {
     resizeObserver.current?.disconnect();
-    if (elementRef.current) {
+    if (element) {
       resizeObserver.current = new ResizeObserver(([{ contentRect }]) =>
```

Comment:
Also, need to check if ResizeObserver API is available.. Not only for IE, but also for client apps tests (so they dont start failing randomly). Check in `useIntersection` hook..Its handled there.
![image](https://user-images.githubusercontent.com/81580355/124760939-eb600300-df39-11eb-8160-3f256aefefc5.png)